app/assets/javascripts/the_bridge_ng/controllers/marketplace_ctrl.js.coffee

```diff
@@ -30,5 +31,24 @@
       return product.integrated
     else
       return true
+
+  $scope.changeVote = (flag, product_id) ->
+    html = $('#num_'+product_id).html()
```

Comment:
We're using angular, jquery style code is unnecessary for this kind of thing.

Rather than working out the state from the DOM, use the product instance that would be passed in from the repeat block.